specification.md

```diff
@@ -319,17 +319,28 @@ Note that different data filters play a big role as to which parts of the states
 evaluated. Reference the 
 [State Data Filtering](#State-Data-Filtering) section for more information about state data filters.
 
-All expressions must follow the [jq](https://stedolan.github.io/jq/) syntax. Jq is a very powerful JSON processor. You 
-can find more information in the [jq manual](https://stedolan.github.io/jq/manual/).
+By default, all workflow expressions should be defined using the [jq](https://stedolan.github.io/jq/) syntax.
+You can find more information on jq in its [manual](https://stedolan.github.io/jq/manual/).
 
-Expressions have the following format:
+Serverless Workflow does not mandate the use of jq and it's possible to use an expression language 
+of your choice with the restriction that a single one must be used for all expressions
+in a workflow definition. If a different expression language needs to be used, make sure to set the workflow 
+`expressionLang` property to identify it to runtime implementations.
```

Comment:
call on 15/02: Do we need to distinguish transformationLang from conditionLang?
jq can do transformation (outputPath, etc.), but when we looked at JSONPath and JSONPatch, path was able to list found elements in an array while JSONPatch was able to make modifications (add/replace/remove/move/copy/...) and we didn't discuss languages that are just for writing conditions, but there are some. We need conditions for transitions, switch statements, etc and we need transformations for input/output path. If I specify expressionLang to be SQL, I wouldn't be able to do any transformations. And for conditions, it needs to be clear how to interpret the result (e.g. if an empty/nil result means false or if it's an error).

Reply:
I'm not sure we can tell users which parts of jq to use at what places, not sure how to honestly enforce that (even on the runtime level). 
a selection in case of our data filters is really also a modification, for example a state filter data output becomes the final output of that state, so by selection you are really modifying the state data that for example is passed to the next state. 

i think for conditions we should just say they are boolean conditions and leave it at that